From da6e58b51cc64b4fe55f1bb92badc147ec386e30 Mon Sep 17 00:00:00 2001
From: Dave Taht <dave.taht@bufferbloat.net>
Date: Fri, 13 Apr 2012 22:55:00 -0700
Subject: [PATCH] UCLIBC does the right thing with strptime and blows up with
 _XOPEN_SOURCE

So add the define that POSIX says to avoid for glibc only.
---
 gpsutils.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/gpsutils.c b/gpsutils.c
index 60c4ae9..966012f 100644
--- a/gpsutils.c
+++ b/gpsutils.c
@@ -4,9 +4,11 @@
  * BSD terms apply: see the file COPYING in the distribution root for details.
  */
 
-/* The strptime prototype is not provided unless explicitly requested.
- *  So add the define that POSIX says to to avoid: */
+/* Glibc does not provide the strptime prototype explicitly requested.
+ *  So add the define that POSIX says to to avoid, but skip it for uclibc. */
+#if !defined(__UCLIBC__)
 #define _XOPEN_SOURCE
+#endif
 
 #include <stdio.h>
 #include <time.h>
-- 
1.7.5.4

